### DISCUSSION TITLE
Weekly report and notifications

### DISCUSSION THREAD

**Discussion by oliverk71** [mcguirepr89/BirdNET-Pi#868] [General]
Notifications work in general, but until now I never got a weekly report via notifications. I also noticed that the time frame is from sunday to saturdary, but here in Europe it should be from monday to sunday. This does not matter much, just to inform you. But I would like to get the weekly report as notification, if possible.

**Comment by Svardsten53** [comment]
I activated the weekly report a week ago, but haven't received one either.
The problem of the week starting with a Monday in Europe can be solved as follows,
The following code at the beginning of weekly_report.php

`$startdate = strtotime('last sunday') - (7*86400);`
`$enddate = strtotime('last sunday') - (1*86400);`

change to

`$startdate = strtotime('last sunday') - (6*86400);`
`$enddate = strtotime('last sunday') - (0*86400);`

I haven't tested this for a full week yet, but so far it works the way I want it to.

I live in Sweden and have by running some scripts after each program update done a lot of national adaptations, including translations and adaptations of external links. BirdNET-Pi is an absolutely wonderful program, but may need some customization if you live in a non-English speaking country.

**Comment by ehpersonal38** [comment]
Can you manually open the following URL and see if this script works?

`http://birdnetpi.local/views.php?view=Weekly%20Report&ascii=true`

You may need to adjust `birdnetpi.local` to your Pi's IP address, if the `.local` hostname system doesn't work on your network.

**Reply by oliverk71** [reply]
Yes, this gives me the weekly report.

**Reply by ehpersonal38** [reply]
Try running `bash ~/BirdNET-Pi/scripts/weekly_report.sh` and check for any errors

**Reply by jmtuts** [reply]
ehpersonal38 - Are there any other views similar to the Weekly view you mention above?  What other parameters can we use on views.php?

**Reply by ehpersonal38** [reply]
@jmtuts No that's really the only "hidden" page, although it's not hidden anymore as per a recent update that added a link to it in Tools -> Settings -> Notifications -> Weekly Report. 
**EDIT:** there actually is another one, it's `views.php?view=Kiosk` which is a kiosk mode I made for someone who never ended up using it. It's basically just the "Today's Detections" page with a few modifications to 1) make it auto refresh the detections every minute or so, and 2) to prevent the user from navigating off the page by disabling all links.
**EDIT 2:** I just tested Kiosk and discovered it's broken, I pushed an update if you want to try it.

As for `views.php`, the parameters it takes can be found here: https://github.com/mcguirepr89/BirdNET-Pi/blob/main/homepage/views.php#L139. It's nothing too interesting, just the page navigation and the occasional shell command (for updating, starting services, etc).

**Reply by oliverk71** [reply]
Here is the output: 

```
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  3929    0  3929    0     0    119      0 --:--:--  0:00:32 --:--:--  1074
2023-04-27 13:45:35,593 - INFO - Loaded 1 entries from file:///home/oliver/BirdNET-Pi/apprise.txt?encoding=utf-8&cache=yes
2023-04-27 13:45:35,605 - INFO - Notifying 1 service(s) asynchronously.
2023-04-27 13:45:38,431 - INFO - Sent [01/01] Mastodon notification as public toot.

```

And I got the weekly report as notification in mastodon for the very first time.

**Reply by oliverk71** [reply]
Need to say that I often have problems reaching the pi at all. This may be due to moisture in the antenna, which was not made for outdoor use. However, if reachable, the birdnetpi is often processing backlogs and sometimes data of one or several days is gone. For example, I do not have any 'detections of the day' from 22. to 24. of May. Right now it is working fine.

**Reply by ehpersonal38** [reply]
If that script is working then I guess it's not getting called from the crontab? Can you please paste the output of the following command?

`cat /etc/crontab`

Note that there should be a blank line at the very bottom of the output of that command. For example, this is my output:
```bash
pi@birdnetpi:~/BirdNET-Pi$ cat /etc/crontab
# /etc/crontab: system-wide crontab
# Unlike any other crontab you don't have to run the `crontab'
# command to install the new version when you edit this file
# and files in /etc/cron.d. These files also have username fields,
# that none of the other crontabs do.

SHELL=/bin/sh
PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin

# Example of job definition:
# .---------------- minute (0 - 59)
# |  .------------- hour (0 - 23)
# |  |  .---------- day of month (1 - 31)
# |  |  |  .------- month (1 - 12) OR jan,feb,mar,apr ...
# |  |  |  |  .---- day of week (0 - 6) (Sunday=0 or 7) OR sun,mon,tue,wed,thu,fri,sat
# |  |  |  |  |
# *  *  *  *  * user-name command to be executed
17 *    * * *   root    cd / && run-parts --report /etc/cron.hourly
25 6    * * *   root    test -x /usr/sbin/anacron || ( cd / && run-parts --report /etc/cron.daily )
47 6    * * 7   root    test -x /usr/sbin/anacron || ( cd / && run-parts --report /etc/cron.weekly )
52 6    1 * *   root    test -x /usr/sbin/anacron || ( cd / && run-parts --report /etc/cron.monthly )
#birdnet
*/5 * * * * pi /usr/local/bin/disk_check.sh &> /dev/null
#birdnet
*/3 * * * * pi /usr/local/bin/cleanup.sh &> /dev/null
#birdnet
@reboot pi /usr/local/bin/cleanup.sh &> /dev/null
#birdnet
0 7 * * 1 pi /usr/local/bin/weekly_report.sh >/dev/null 2>&1
                                                                                 
pi@birdnetpi:~/BirdNET-Pi$
```

**Reply by jmtuts** [reply]
@ehpersonal38 - I updated and tried the Kiosk view, just get a blank screen.  Is this the correct URL for that view?
http://birdnet.local/views.php?view=Kiosk%20Report&ascii=true

**Reply by ehpersonal38** [reply]
It's http://birdnetpi.local/views.php?view=Kiosk

**Reply by oliverk71** [reply]
@ehpersonal38 Here is my output. Spoiler: no blank line at the bottom.

```
birdnetpi-elsen:~/BirdNET-Pi$ cat /etc/crontab
# /etc/crontab: system-wide crontab
# Unlike any other crontab you don't have to run the `crontab'
# command to install the new version when you edit this file
# and files in /etc/cron.d. These files also have username fields,
# that none of the other crontabs do.

SHELL=/bin/sh
PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin

# Example of job definition:
# .---------------- minute (0 - 59)
# |  .------------- hour (0 - 23)
# |  |  .---------- day of month (1 - 31)
# |  |  |  .------- month (1 - 12) OR jan,feb,mar,apr ...
# |  |  |  |  .---- day of week (0 - 6) (Sunday=0 or 7) OR sun,mon,tue,wed,thu,fri,sat
# |  |  |  |  |
# *  *  *  *  * user-name command to be executed
17 *	* * *	root    cd / && run-parts --report /etc/cron.hourly
25 6	* * *	root	test -x /usr/sbin/anacron || ( cd / && run-parts --report /etc/cron.daily )
47 6	* * 7	root	test -x /usr/sbin/anacron || ( cd / && run-parts --report /etc/cron.weekly )
52 6	1 * *	root	test -x /usr/sbin/anacron || ( cd / && run-parts --report /etc/cron.monthly )
#
#birdnet
*/5 * * * * oliver /usr/local/bin/disk_check.sh >/dev/null 2>&1
#birdnet
*/3 * * * * oliver /usr/local/bin/cleanup.sh >/dev/null 2>&1
#birdnet
@reboot oliver /usr/local/bin/cleanup.sh >/dev/null 2>&1
#birdnet
0 1 * * 1 oliver /usr/local/bin/weekly_report.sh >/dev/null 2>&1
birdnetpi-elsen:~/BirdNET-Pi$ 
```

**Reply by ehpersonal38** [reply]
Thanks, can I see the output of this command?
`grep CRON /var/log/syslog | tail -n 100`

**Reply by oliverk71** [reply]
Oh, I did not see this! Unfortunately, my installation has far bigger problems, now.

**Comment by jmtuts** [comment]
@ehpersonal38 Thank you, that worked
